src/test/java/com/es/phoneshop/service/implementation/ProductServiceImplTest.java

```diff
@@ -0,0 +1,161 @@
+package com.es.phoneshop.service.implementation;
+
+import com.es.phoneshop.dao.ProductDao;
+import com.es.phoneshop.dao.implementation.ArrayListProductDaoImpl;
+import com.es.phoneshop.exception.ProductNotFoundException;
+import com.es.phoneshop.model.Product;
+import com.es.phoneshop.service.ProductService;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+
+import java.math.BigDecimal;
+import java.util.Currency;
+import java.util.List;
+import java.util.UUID;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+public class ProductServiceImplTest {
+    private ProductService productService;
+    private Currency usd;
+
+    @Mock
+    private ProductDao productDao;
+
+    @Before
+    public void setup() {
+        usd = Currency.getInstance("USD");
+        productService = new ProductServiceImpl();
+        productDao = new ArrayListProductDaoImpl();
+    }
+
+    @Test
+    public void getProduct_ValidId_ShouldReturnProductWithIdFromDao() {
+        // given
+        final UUID id = productService.findProducts().get(0).getId();
+        final BigDecimal expectedPrice = new BigDecimal(100);
+        final String expectedDescription = "Samsung Galaxy S";
+        final int expectedStock = 100;
+
+        // when
+        Product product = productService.getProduct(id);
```

Comment:
In this case it is enough to test that product service calls productDao.getProduct(id) method via Mockito verify() method, the rest of the logic is the responsibility of productDao you are testing in another test class

Reply:
the same for all other tests